cosmos/dbt/project.py

```diff
@@ -6,15 +6,22 @@
 from pathlib import Path
 from typing import Generator
 
-from cosmos.constants import DBT_LOG_DIR_NAME, DBT_PARTIAL_PARSE_FILE_NAME, DBT_TARGET_DIR_NAME
+from cosmos.constants import (
+    DBT_LOG_DIR_NAME,
+    DBT_PACKAGES_CONFIG_PATH,
+    DBT_PARTIAL_PARSE_FILE_NAME,
+    DBT_TARGET_DIR_NAME,
+)
 
 
 def create_symlinks(project_path: Path, tmp_dir: Path, ignore_dbt_packages: bool) -> None:
     """Helper function to create symlinks to the dbt project files."""
     ignore_paths = [DBT_LOG_DIR_NAME, DBT_TARGET_DIR_NAME, "profiles.yml"]
-    if ignore_dbt_packages:
-        # this is linked to dbt deps so if dbt deps is true then ignore existing dbt_packages folder
+
+    if ignore_dbt_packages and DBT_PACKAGES_CONFIG_PATH.exists():
```

Comment:
To create a test to validate this change and make sure we don't miss this use-case again, we could have a fixture similar to:
https://github.com/astronomer/astronomer-cosmos/blob/ddb25cbbcb6de3bde9d784b5fa59786c5b3beae8/tests/dbt/test_graph.py#L31-L45

And a test similar to:
https://github.com/astronomer/astronomer-cosmos/blob/ddb25cbbcb6de3bde9d784b5fa59786c5b3beae8/tests/dbt/test_project.py#L12-L20